node/src/client/mod.rs

```diff
@@ -277,7 +284,14 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
         if block_requests.is_empty() && self.sync.has_pending_responses() {
             // Try to advance the ledger with the sync pool.
             trace!("No block requests to send, but there are still pending block responses.");
-            self.sync.try_advancing_block_synchronization();
+            let has_new_blocks = self.sync.try_advancing_block_synchronization();
```

Comment:
Is it worth / possible to call `try_block_sync` on receiving new block locators, instead of waiting a fixed 5 seconds?

Reply:
Yes, that is another thing I was thinking of doing. We should trigger try_advancing_block_synchronization, not at a fixed interval but on specific events. It should just be the following
* New block locators were received
* All outstanding block requests were processed
* A peer disconnects (can affect outstanding block requests)

We can still keep the timer as a failsafe. I thought that might be better as a separate PR, but I can add some of it to this one.

Reply:
Made an issue for it https://github.com/ProvableHQ/snarkOS/issues/3691